File: frontend/src/types/customer.ts
export type CustomerType = "Employer" | "Employee" | "Both";

export interface Customer {
  id: number;
  user_id: number;
  name: string;
  birth_date: string | null;
  phone: string;
  address: string | null;
  profile_photo_id: number | null;
  customer_type: CustomerType;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface CustomerMemo {
  id: number;
  customer_id: number;
  memo_content: string;
  created_by: number | null;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface CustomerFile {
  id: number;
  customer_id: number;
  file_path: string;
  file_type: string;
  file_size: number | null;
  thumbnail_path: string | null;
  original_filename: string | null;
  mime_type: string | null;
  is_profile: boolean;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface CustomerTag {
  id: number;
  user_id: number;
  tag_name: string;
  tag_color: string;
  description: string | null;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface CreateCustomerRequest {
  name: string;
  birth_date?: string | null;
  phone: string;
  address?: string | null;
  customer_type: CustomerType;
}

export interface UpdateCustomerRequest {
  name?: string;
  birth_date?: string | null;
  phone?: string;
  address?: string | null;
  profile_photo_id?: number | null;
  customer_type?: CustomerType;
}

File: backend/src/models/customer.rs
use chrono::{NaiveDate, NaiveDateTime};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type, PartialEq)]
#[sqlx(type_name = "text", rename_all = "lowercase")]
#[serde(rename_all = "PascalCase")]
pub enum CustomerType {
    #[sqlx(rename = "employer")]
    Employer,
    #[sqlx(rename = "employee")]
    Employee,
    #[sqlx(rename = "both")]
    Both,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Customer {
    pub id: i64,
    pub user_id: i64,
    pub name: String,
    pub birth_date: Option<NaiveDate>,
    pub phone: String,
    pub address: Option<String>,
    pub profile_photo_id: Option<i64>,
    pub customer_type: CustomerType,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
    pub deleted_at: Option<NaiveDateTime>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CreateCustomerRequest {
    pub name: String,
    pub birth_date: Option<NaiveDate>,
    pub phone: String,
    pub address: Option<String>,
    pub customer_type: CustomerType,
}

#[derive(Debug, Clone, Deserialize)]
pub struct UpdateCustomerRequest {
    pub name: Option<String>,
    pub birth_date: Option<NaiveDate>,
    pub phone: Option<String>,
    pub address: Option<String>,
    pub profile_photo_id: Option<i64>,
    pub customer_type: Option<CustomerType>,
}
